src/distributions/other.rs

```diff
@@ -85,6 +89,19 @@ impl Distribution<char> for Standard {
     }
 }
 
+/// Note: the `String` is potentially left with excess capacity; optionally the
+/// user may call `string.shrink_to_fit()` afterwards.
+#[cfg(feature = "alloc")]
+impl DistString for Standard {
+    fn append_string<R: Rng + ?Sized>(&self, rng: &mut R, s: &mut String, len: usize) {
+        // A char is encoded with at most four bytes, thus this reservation is
+        // guaranteed to be sufficient. We do not shrink_to_fit afterwards so
+        // that repeated usage on the same `String` buffer does not reallocate.
+        s.reserve(s.len() + 4 * len);
```

Comment:
Unnecessarily doubles the buffer. The parameter to `reserve` is the count of additional elements.